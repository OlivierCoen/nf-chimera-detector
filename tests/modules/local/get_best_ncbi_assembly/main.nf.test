nextflow_process {

    name "Test Process GET_BEST_NCBI_ASSEMBLY"
    script "modules/local/get_best_ncbi_assembly/main.nf"
    process "GET_BEST_NCBI_ASSEMBLY"
    tag "get_best_assembly"

    test("Many assemblies available") {

        when {
            process {
                """
                input[0] = [
                    [ ],
                    558016
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            assert process.out.accession == [[[], 'GCF_000870065.1']]
        }

    }

    test("No assembly available") {

        when {
            process {
                """
                input[0] = [
                    [ ],
                    307457
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            assert process.out.accession == [[[], 'NONE']]
        }

    }

}
