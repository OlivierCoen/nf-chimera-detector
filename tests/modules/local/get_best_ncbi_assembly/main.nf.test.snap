{
    "Many assemblies available": {
        "content": [
            {
                "0": [
                    [
                        [
                            
                        ],
                        "GCF_000870065.1"
                    ]
                ],
                "1": [
                    [
                        [
                            
                        ],
                        "assembly_report.json:md5,7709c25828c810c2eb31e19f7ed679e5"
                    ]
                ],
                "2": [
                    [
                        "GET_BEST_NCBI_ASSEMBLY",
                        "python",
                        "3.13.5"
                    ]
                ],
                "3": [
                    [
                        "GET_BEST_NCBI_ASSEMBLY",
                        "requests",
                        "2.32.4"
                    ]
                ],
                "accession": [
                    [
                        [
                            
                        ],
                        "GCF_000870065.1"
                    ]
                ],
                "assembly_report": [
                    [
                        [
                            
                        ],
                        "assembly_report.json:md5,7709c25828c810c2eb31e19f7ed679e5"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-24T09:39:50.545950902"
    },
    "No assembly available": {
        "content": [
            {
                "0": [
                    [
                        [
                            
                        ],
                        "NONE"
                    ]
                ],
                "1": [
                    [
                        [
                            
                        ],
                        "assembly_report.json:md5,03ed5bf31c35414f6da7c526c1a67039"
                    ]
                ],
                "2": [
                    [
                        "GET_BEST_NCBI_ASSEMBLY",
                        "python",
                        "3.13.5"
                    ]
                ],
                "3": [
                    [
                        "GET_BEST_NCBI_ASSEMBLY",
                        "requests",
                        "2.32.4"
                    ]
                ],
                "accession": [
                    [
                        [
                            
                        ],
                        "NONE"
                    ]
                ],
                "assembly_report": [
                    [
                        [
                            
                        ],
                        "assembly_report.json:md5,03ed5bf31c35414f6da7c526c1a67039"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-24T09:39:59.643510077"
    }
}